receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -349,7 +349,11 @@ func (prw *prometheusRemoteWriteReceiver) translateV2(_ context.Context, req *wr
 		case writev2.Metadata_METRIC_TYPE_COUNTER:
 			addNumberDatapoints(metric.Sum().DataPoints(), ls, ts)
 		case writev2.Metadata_METRIC_TYPE_HISTOGRAM:
-			addHistogramDatapoints(metric.Histogram().DataPoints(), ls, ts)
+			if len(ts.Samples) > 0 {
+				badRequestErrors = errors.Join(badRequestErrors, errors.New("this flow just support the conversion of native histograms to exponential histograms. classic histograms will be dropped"))
```

Comment:
This makes me a little bit nervous from performance regression point of view as remote write requests can have a lot of classic histograms and errors.Join will allocate arrays and append all the time here. I'd suggest to introduce an `error` type variable that can tell in the end whether we ran into this error or not. Or just not do anything as the current code does afaik.

Reply:
[f531196](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39864/commits/f531196da0eb30218c84b148a343e4782043dcbc)